### PR TITLE
Metadata Refactor

### PR DESCRIPTION
**Pull request by jbouffard**
This Pr is based another, open Pr #84 . This Pr seeks to add a new way of collecting metadata through using `FloatingLayoutScheme`. In addition to this new method, the metadata collecting code has been cleaned up and a few function names have been changes.

### CODE REVIEW

**Comment by jbouffard** [comment]
@echeipesh This is ready for review once #84 is merged.

File: geopyspark/geotrellis/tile_layer.py [hunk 1]
```diff
@@ -233,17 +233,48 @@ def collect_pyramid_metadata(geopysc,
...
+                                                    schema,
+                                                    crs,
+                                                    tile_size,
+                                                    resolution_threshold,
+                                                    max_zoom,
+                                                    output_crs)
 
     return (result._1(), json.loads(result._2()))
 
+def collect_pyramid_floating_metadata(geopysc,
```
Comment (echeipesh) [inline geopyspark/geotrellis/tile_layer.py]: `pyramid` and `floating` are mutually exclusive. If layer is part of a pyramid, it must be one of the power of two levels with corresponding resolution. Floating layouts are for layouts with resolutions and tile layouts that don't match any pyramid level, but rather match the source data as closely as possible.

File: geopyspark/geotrellis/tile_layer.py [hunk 2]
```diff
@@ -129,14 +129,14 @@ def collect_metadata(geopysc,
...
 
-def collect_pyramid_metadata(geopysc,
-                             rdd_type,
-                             raster_rdd,
-                             crs,
-                             tile_size,
-                             resolution_threshold=0.1,
-                             max_zoom=12,
-                             output_crs=None):
+def collect_pyramid_zoomed_metadata(geopysc,
```
Comment (echeipesh) [inline geopyspark/geotrellis/tile_layer.py]: What is the addition of `zoomed` meant to communicate ?

**Comment by jbouffard** [comment]
@echeipesh The reason I gave the functions those names at the time was because I had an incorrect understanding of their usage. Now that I know the distinctions between them, I've corrected the issue.

**Pull request merged**